libs/language-server/src/lib/ast/wrappers/pipe-wrapper.ts

```diff
@@ -9,76 +9,42 @@ import { DiagnosticInfo } from 'langium';
 import {
   BlockDefinition,
   BlocktypePipeline,
-  ChainedPipeDefinition,
   PipeDefinition,
-  SinglePipeDefinition,
-  isSinglePipeDefinition,
 } from '../generated/ast';
 
 import { AstNodeWrapper } from './ast-node-wrapper';
 
-export class PipeWrapper<N extends PipeDefinition = PipeDefinition>
-  implements AstNodeWrapper<N>
+export class PipeWrapper
+  implements AstNodeWrapper<PipeDefinition | BlocktypePipeline>
```

Comment:
Why did you remove the generic? `astNode` can be narrowed down with the generic.

 ```ts
export class PipeWrapper<T extends PipeDefinition | BlocktypePipeline>
  implements AstNodeWrapper<T> { ... }
```

Current State:
```ts
declare const pipeline: BlocktypePipeline;
const wrapper = new PipeWrapper(pipeline, 29);
const node = wrapper.astNode; // <- typed as 'PipeDefinition | BlocktypePipeline' instead of 'BlocktypePipeline'
```